src/pages/iou/IOUCurrencySelection.js

```diff
@@ -0,0 +1,239 @@
+     */
+    getCurrencyOptions() {
+        const currencyListKeys = _.keys(this.props.currencyList);
+        const currencyOptions = _.map(currencyListKeys, currencyCode => ({
+            text: `${currencyCode} - ${this.props.currencyList[currencyCode].symbol}`,
+            searchText: `${currencyCode} ${this.props.currencyList[currencyCode].symbol}`,
+            currencyCode,
+        }));
+        return currencyOptions;
+    }
+
+    /**
+     * Function which renders a row in the list
+     *
+     * @param {String} currencyCode
+     *
+     */
+    toggleOption(currencyCode) {
+        this.setState({
+            selectedCurrency: {
+                currencyCode,
+                currencySymbol: this.props.currencyList[currencyCode].symbol,
+            },
+        });
+    }
+
+    /**
+     * Returns the key used by the list
+     * @param {Object} option
+     * @return {String}
+     */
+    extractKey(option) {
+        return option.currencyCode;
+    }
+
+    /**
+     * Function which renders a row in the list
+     *
+     * @param {Object} params
+     * @param {Object} params.item
+     *
+     * @return {Component}
+     */
+    renderItem({item, key}) {
+        return (
+            <OptionRow
+                key={key}
+                mode="compact"
+                option={item}
+                onSelectRow={() => this.toggleOption(item.currencyCode)}
+                isSelected={item.currencyCode === this.state.selectedCurrency.currencyCode}
+                showSelectedState
+                hideAdditionalOptionStates
+            />
+        );
+    }
+
+    render() {
+        const sections = this.getSections();
+        return (
+            <ScreenWrapper>
+                <HeaderWithCloseButton
+                    title="Select a currency"
+                    onCloseButtonPress={() => Navigation.goBack()}
+                />
+                <View style={[styles.flex1, styles.w100]}>
+                    <View style={[styles.flex1]}>
+                        <View style={[styles.ph5, styles.pv3]}>
+                            <TextInputWithFocusStyles
+                                styleFocusIn={[styles.textInputReversedFocus]}
+                                ref={el => this.textInput = el}
+                                style={[styles.textInput]}
+                                value={this.state.searchValue}
+                                onChangeText={(searchValue = '') => {
+                                    const {currencyOptions} = getCurrencyListForSections(
+                                        this.getCurrencyOptions(this.props.currencyList),
+                                        searchValue,
+                                    );
+                                    this.setState({
+                                        searchValue,
+                                        currencyData: currencyOptions,
+                                    });
+                                }}
```

Comment:
I think this would be better off in a class-scoped function, then you can just pass it like

```javascript
onChangeText={this.nameOfFunction}
```

Since it's not entirely clear to me what this function does at first glance, it'd be good to give `nameOfFunction` a descriptive name.